Source/Fluids/WarpXFluidContainer.cpp

```diff
@@ -0,0 +1,1854 @@
+                    NUz_arr(i,j,k) = NUz_arr(i,j,k-2);
+
+                // Lower end (index 2)
+                } else if ( (periodic_directions[2] != 1) && (k==domain.smallEnd(2)-1) ) {
+                    N_arr(i,j,k) = N_arr(i,j,k+2);
+                    NUx_arr(i,j,k) = NUx_arr(i,j,k+2);
+                    NUy_arr(i,j,k) = NUy_arr(i,j,k+2);
+                    NUz_arr(i,j,k) = NUz_arr(i,j,k+2);
+                }
+
+#elif ( defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ) || defined(WARPX_DIM_3D) )
+
+                // Upper end (index 1)
+                if ( (periodic_directions[1] != 1) && (j==domain.bigEnd(1)+1) ){
+                    N_arr(i,j,k) = N_arr(i,j-2,k);
+                    NUx_arr(i,j,k) = NUx_arr(i,j-2,k);
+                    NUy_arr(i,j,k) = NUy_arr(i,j-2,k);
+                    NUz_arr(i,j,k) = NUz_arr(i,j-2,k);
+
+                // Lower end (index 1`)
+                } else if ( (periodic_directions[1] != 1) && (j==domain.smallEnd(1)-1) ) {
+                    N_arr(i,j,k) = N_arr(i,j+2,k);
+                    NUx_arr(i,j,k) = NUx_arr(i,j+2,k);
+                    NUy_arr(i,j,k) = NUy_arr(i,j+2,k);
+                    NUz_arr(i,j,k) = NUz_arr(i,j+2,k);
+
+                }
+
+#elif ( defined(WARPX_DIM_1D_Z) || defined(WARPX_DIM_XZ) || defined(WARPX_DIM_3D) )
+
+                // Upper end (index 0)
+                if ( (periodic_directions[0] != 1) && (i==domain.bigEnd(0)+1) ){
+                    N_arr(i,j,k) = N_arr(i-2,j,k);
+                    NUx_arr(i,j,k) = NUx_arr(i-2,j,k);
+                    NUy_arr(i,j,k) = NUy_arr(i-2,j,k);
+                    NUz_arr(i,j,k) = NUz_arr(i-2,j,k);
+
+                // Lower end (index 0)
+                } else if ( (periodic_directions[0] != 1) && (i==domain.smallEnd(0)-1) ) {
+                    N_arr(i,j,k) = N_arr(i+2,j,k);
+                    NUx_arr(i,j,k) = NUx_arr(i+2,j,k);
+                    NUy_arr(i,j,k) = NUy_arr(i+2,j,k);
+                    NUz_arr(i,j,k) = NUz_arr(i+2,j,k);
+                }
+
+#else
+
+#endif
+            }
+        );
+    }
+
+    // Fill guard cells
+    FillBoundary(*N[lev], N[lev]->nGrowVect(), WarpX::do_single_precision_comms, period);
+    FillBoundary(*NU[lev][0], NU[lev][0]->nGrowVect(), WarpX::do_single_precision_comms, period);
+    FillBoundary(*NU[lev][1], NU[lev][1]->nGrowVect(), WarpX::do_single_precision_comms, period);
+    FillBoundary(*NU[lev][2], NU[lev][2]->nGrowVect(), WarpX::do_single_precision_comms, period);
+}
+
+// Muscl Advection Update
+void WarpXFluidContainer::AdvectivePush_Muscl (int lev)
+{
+    WARPX_PROFILE("WarpXFluidContainer::AdvectivePush_Muscl");
+
+    // Grab the grid spacing
+    WarpX &warpx = WarpX::GetInstance();
+    const Real dt = warpx.getdt(lev);
+    const amrex::Geometry &geom = warpx.Geom(lev);
+    const auto dx = geom.CellSizeArray();
+    const amrex::Real clight = PhysConst::c;
+#if defined(WARPX_DIM_3D)
+    amrex::Real cx = (dt/dx[0]);
```

Comment:
Could you use a more explicity name for this variable:
```suggestion
    amrex::Real dt_over_dx = (dt/dx[0]);
```

Reply:
As well as the other below...

Reply:
Fixed.